core/services/offchainreporting/OCRContract.go

```diff
@@ -0,0 +1,232 @@
+package offchainreporting
+
+import (
+	"context"
+	"math/big"
+	"strings"
+	"sync"
+	"time"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	gethCommon "github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services/eth"
+	"github.com/smartcontractkit/chainlink/core/store/models"
+	"github.com/smartcontractkit/chainlink/offchainreporting/confighelper"
+	"github.com/smartcontractkit/chainlink/offchainreporting/gethwrappers/offchainaggregator"
+	ocrtypes "github.com/smartcontractkit/chainlink/offchainreporting/types"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/pkg/errors"
+)
+
+var (
+	OCRContractConfigSet = getConfigSetHash()
+)
+
+type (
+	OCRContract struct {
+		ethClient        eth.Client
+		contractFilterer *offchainaggregator.OffchainAggregatorFilterer
+		contractCaller   *offchainaggregator.OffchainAggregatorCaller
+		contractAddress  gethCommon.Address
+		logBroadcaster   eth.LogBroadcaster
+		jobID            models.ID
+		transmitter      Transmitter
+		contractABI      abi.ABI
+		logger           logger.Logger
+	}
+
+	Transmitter interface {
+		CreateEthTransaction(ctx context.Context, toAddress gethCommon.Address, payload []byte) error
+		FromAddress() gethCommon.Address
+	}
+)
+
+var (
+	_ ocrtypes.ContractConfigTracker      = &OCRContract{}
+	_ ocrtypes.ContractTransmitter        = &OCRContract{}
+	_ ocrtypes.ContractConfigSubscription = &OCRContractConfigSubscription{}
+	_ eth.LogListener                     = &OCRContractConfigSubscription{}
+)
+
+func NewOCRContract(address gethCommon.Address, ethClient eth.Client, logBroadcaster eth.LogBroadcaster, jobID models.ID, transmitter Transmitter, logger logger.Logger) (o *OCRContract, err error) {
+	contractFilterer, err := offchainaggregator.NewOffchainAggregatorFilterer(address, ethClient)
+	if err != nil {
+		return o, errors.Wrap(err, "could not instantiate NewOffchainAggregatorFilterer")
```

Comment:
Seem to be using a mix of named returns here and explicit?

Reply:
Is it a problem?